mlflow/pyspark/ml/__init__.py

```diff
@@ -57,19 +57,24 @@ def _parse_allowlist_file(line_iter):
         return allowlist
 
     url_parsed = urlparse(allowlist_file)
-    if url_parsed.scheme in ("file", ""):
-        if not os.path.exists(url_parsed.path):
+    scheme = url_parsed.scheme
+    path = url_parsed.path
+    if os.name == "nt" and not url_parsed.hostname:
+        path = scheme + "://" + path
+        scheme = ""
+    if scheme in ("file", ""):
+        if not os.path.exists(path):
             raise MlflowException.invalid_parameter_value(f"{allowlist_file} does not exist")
 
         with open(allowlist_file) as f:
             return _parse_allowlist_file(f)
     else:
         host_creds = MlflowHostCreds(
-            host=url_parsed.scheme + "://" + url_parsed.hostname or "",
+            host=scheme + "://" + (url_parsed.hostname or ""),
```

Comment:
Wait, `http://` is this a valid host ?

Reply:
Normally if the url starts with http:// it should have a reasonable hostname, and if it doesn't the error should be caught in augmented_raise_for_status. Otherwise the error shows strangely here as cannot concatenate string & Nonetype.

Reply:
ok.